registry/storage/driver/azure/azure.go

```diff
@@ -383,6 +397,14 @@ func (d *driver) listBlobs(container, virtPath string) ([]string, error) {
 }
 
 func is404(err error) bool {
+	// handle the case when the request was a HEAD and service error could not
+	// be parsed, such as "storage: service returned without a response body
+	// (404 The specified blob does not exist.)"
+	if strings.Contains(fmt.Sprintf("%v", err), "404 The specified blob does not exist") {
```

Comment:
**NOTE:** we can remove this new conditional once we bring the following patch here by vendoring the latest azure-sdk-for-go https://github.com/Azure/azure-sdk-for-go/pull/480